### PR TITLE
Fix dashboard delete

### PR DESCRIPTION
**Pull request by jmuharsky**
Also adds function-level unit tests to DashboardToolbarDirective.

### CODE REVIEW

File: client/components/dashboard/dashboard-toolbar-directive.js [hunk 1]
```diff
@@ -99,7 +107,7 @@ goog.scope(function() {
          * @export
          */
         this.downloadDashboard = function() {
-          let selectedDashboard = this.dashboardSvc.current.model;
+          var selectedDashboard = this.dashboardSvc.current.model;
```
Comment (cmccoy) [inline client/components/dashboard/dashboard-toolbar-directive.js]: Why change this?

File: client/components/dashboard/dashboard-toolbar-directive.js [hunk 2]
```diff
@@ -120,14 +128,15 @@ goog.scope(function() {
...
-              this.dashboard.current.model);
+          var promise = this.dashboardSvc.deleteDashboard(
+              this.dashboardSvc.current.model);
 
           promise.then(angular.bind(this, function(response) {
             this.openDashboardAdmin();
           }));
 
           promise.then(null, angular.bind(this, function(error) {
+            console.log('failed');
```
Comment (cmccoy) [inline client/components/dashboard/dashboard-toolbar-directive.js]: Either drop the log statement or log a more actionable message (maybe include `error`?).

**Comment by cmccoy** [comment]
Thanks for separating the commits :)

File: client/components/dashboard/dashboard-toolbar-directive.js (hunk 1 above)
Comment (jmuharsky) [inline client/components/dashboard/dashboard-toolbar-directive.js]: I had put them in a test (which still run ES5), and apparently got overzealous in my cleanup.

File: client/components/dashboard/dashboard-toolbar-directive.js (hunk 2 above)
Comment (jmuharsky) [inline client/components/dashboard/dashboard-toolbar-directive.js]: Mistakenly left in whilst trying to debug testing a promise.  Removed the logging.

**Comment by jmuharsky** [comment]
np.  PTAL?

File: client/components/dashboard/dashboard-toolbar-directive.js [hunk 3]
```diff
@@ -43,6 +43,15 @@ goog.scope(function() {
         this.tabSvc = sidebarTabService;
 
         /**
+         * Navigates to a new url.
+         * @param {string} url
+         * @export
+         */
+        this.setWindowLocation = function(url) {
```
Comment (cmccoy) [inline client/components/dashboard/dashboard-toolbar-directive.js]: Nit: does this need to be public / exported?

**Comment by cmccoy** [comment]
LGTM.

File: client/components/dashboard/dashboard-toolbar-directive.js (hunk 3 above)
Comment (jmuharsky) [inline client/components/dashboard/dashboard-toolbar-directive.js]: For testability purposes, yes.

**Pull request merged**